reservations/api.py

```diff
@@ -1,13 +1,45 @@
 from rest_framework import mixins, permissions, renderers, serializers, viewsets
 
-from .models import Reservation
+from harbors.models import BoatType, Harbor
+from .models import HarborChoice, Reservation
+
+
+class HarborChoiceSerializer(serializers.ModelSerializer):
+    harbor = serializers.SlugRelatedField(
+        queryset=Harbor.objects.all(),
+        slug_field='identifier'
+    )
+
+    class Meta:
+        model = HarborChoice
+        fields = ('priority', 'harbor',)
 
 
 class ReservationSerializer(serializers.ModelSerializer):
+    chosen_harbors = HarborChoiceSerializer(source='harborchoice_set', many=True)
+    boat_type = serializers.SlugRelatedField(
+        queryset=BoatType.objects.all(),
+        slug_field='identifier'
+    )
+
     class Meta:
         model = Reservation
         fields = '__all__'
 
+    def create(self, validated_data):
+        # Django REST requires to explicitly write creation logic
+        # for ManyToMany relations with a Through model.
+        chosen_harbors = validated_data.pop('harborchoice_set')
+        reservation = super(ReservationSerializer, self).create(validated_data)
```

Comment:
Just `super().create(...)` is enough in Python 3.

Reply:
ah, right! forgot that those could be ommited. 👍